yellowbox/extras/logstash.py

```diff
@@ -1,30 +1,263 @@
-from typing import Optional
-
-from docker import DockerClient
+import json
+import logging
+import selectors
+import socket
+import threading
+from typing import Any, Callable, Dict, Iterable, List, Optional, Union, cast, TypedDict
+from weakref import WeakMethod
 
 from yellowbox.retry import RetrySpec
-from yellowbox.containers import get_ports, create_and_pull
-from yellowbox.subclasses import SingleContainerService, RunMixin
+from yellowbox.subclasses import YellowService
+from yellowbox.utils import _docker_host_name
+
+__all__ = ['LogstashService']
+_logger = logging.getLogger(__name__)
+
+_LISTEN_BACKLOG = 5  # Awaiting connections backlog count
+_STOP_TIMEOUT = 5  # Timeout for stopping the service
+
+# Sent on internal connection to signal closing of background thread
+_CLOSE_SENTINEL = b"\0"
+
+
+class LogstashService(YellowService):
```

Comment:
Since Logstash is a known technology that we only emulate and do not actually run, I think this service should have a different name.

Reply:
Bring one up. Not @aviramha, nor me managed to think of one.

Reply:
`LoggingService`, `LoggerService`, `LogService`

Reply:
Non of these are descriptive for "logs sent in json_lines format" .

Reply:
Oh and the logs are formatted in their special logstash format as well, with `@` and stuff.